hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotCache.java

```diff
@@ -221,11 +213,24 @@ public void release(OmSnapshot omSnapshot) {
     release(snapshotTableKey);
   }
 
+  /**
+   * Callback method used to enqueue or dequeue ReferenceCounted from
+   * pendingEvictionList.
+   * @param referenceCounted ReferenceCounted object
+   */
+  @Override
+  public void callback(ReferenceCounted referenceCounted) {
+    if (referenceCounted.getTotalRefCount() == 0L) {
+      // Reference count reaches zero, add to pendingEvictionList
+      pendingEvictionQueue.add((String) referenceCounted.getParentKey());
+    }
+  }
+
   /**
    * Wrapper for cleanupInternal() that is synchronized to prevent multiple
    * threads from interleaving into the cleanup method.
    */
-  private synchronized void cleanup() {
+  public synchronized void cleanup() {
```

Comment:
nit:
1. I think we can remove `synchronized` now.
2. `cleanupInternal` doesn't have to be separate function. We can remove it and move the code to `cleanup`.

Reply:
done